pkg/keyidentifier/testdata/make.sh

```diff
@@ -3,97 +3,170 @@
 # This script uses several tools to create ssh keys. We attempt to be
 # as exhaustive as possible to create a wide range of things to test.
 #
-# This puts them into a somewhat ugly filesystem hierarchy. The intent
-# is to group things to make it simpler for tools to test
-# groups. The layout should straightforward.
-
-
+# each function also generates a json spec file that describes various
+# attributes of a key, plus the name of the key described. This spec file is
+# consumed by golang tests, and the output of the keyidentifier package is
+# compared against these expected values.
 
 set -e
 #set -x
 
 function rand {
-    dd if=/dev/random bs=1 count=16 2> /dev/null  | base64
+    dd if=/dev/random bs=1 count=16 2> /dev/null  | base64 | sed 's/[^0-9a-zA-Z]//g'
 }
 
-# My openssh is so new, it only makes the new format
-function makeOpensshKey {
+DATA_DIR="specs"
+
+# generate some keys and the corresponding json spec files for testing fingerprints
+function makeOpensshKeyAndSpec {
     type=$1
     bits=$2
-    os="openssh7"
-    dir_fragment="$type/$bits/openssh-new"
-    mkdir -p {encrypted,plaintext}"/$dir_fragment"
-    ssh-keygen -t $type -b $bits  -C "" -f "plaintext/$dir_fragment/seph-$os" -P ""
-    ssh-keygen -t $type -b $bits  -C "" -f "encrypted/$dir_fragment/seph-$os" -P "$(rand)"
-}
+    encrypted=$3
+    format="openssh7"
+    source="openssh"
 
-makeOpensshKey rsa 1024
-makeOpensshKey rsa 2048
-makeOpensshKey rsa 4096
+    keyfile=$(rand)
+    keypath="$DATA_DIR/$keyfile"
 
-makeOpensshKey dsa 1024
 
-makeOpensshKey ecdsa 256
-makeOpensshKey ecdsa 521
+    if [ $encrypted == true ]; then
+        ssh-keygen -t $type -b $bits  -C "" -f $keypath -P "$(rand)"
+        cmd='ssh-keygen -t $type -b $bits  -C \"\" -f $keypath -P $(rand)'
+    else
+        ssh-keygen -t $type -b $bits  -C "" -f $keypath -P ""
+        cmd='ssh-keygen -t $type -b $bits  -C \"\" -f $keypath -P \"\"'
+    fi
 
-# ed25519 is fixed legth, bits is ignored
-makeOpensshKey ed25519 2048
+    fingerprint=$(ssh-keygen -l -f $keypath | awk '{print $2}')
+    md5fingerprint=$(ssh-keygen -l -E md5 -f $keypath | awk '{print $2}' | sed 's/^MD5://')
+
+    cat <<EOF > $keypath.json
+    {
+      "ExpectedFingerprintSHA256": "$fingerprint",
+      "ExpectedFingerprintMD5": "$md5fingerprint",
+      "Type": "$type",
+      "Bits": $bits,
+      "Encrypted": $encrypted,
+      "KeyPath": "$keyfile",
```

Comment:
Shouldn't need this. The name is always strings.TrimSuffix(".json", specFilePath)

Reply:
I thought we were moving _away_ from 'test data encoded in filepath' 😛 

I don't _really_ have a strong opinion here, but it seems more consistent to encode everything in the json spec file.

Reply:
I guess I can go either way here. It's not really test _data_ per say. The existing of `key` and `key.json` define the test. It's never right to have one without the other...

Reply:
that's a fair assessment, and it simplifies the make script as well